datafusion/src/datasource/datasource.rs

```diff
@@ -108,6 +108,11 @@ pub trait TableProvider: Sync + Send {
     /// Statistics should be optional because not all data sources can provide statistics.
     fn statistics(&self) -> Statistics;
 
+    /// Returns whether statistics provided are exact values or estimates
```

Comment:
The nice thing about adding a `has_exact_statistics` is that it is a backwards compatible API

An alternate might be to encapsulate the "Exact statistics or not" into a field on `Statistics` itself, which feels to me like it keeps related things together more, but has the downside of changing Statistics / APIs

Reply:
That's a similar thought process I had. 
Maybe at some point it would also be nice to tell what parts of the statistics are exact (e.g. number of rows) and what estimated (such as distinct count).